downloader/src/main/kotlin/Downloader.kt

```diff
@@ -238,9 +238,11 @@ class Downloader {
             //       non-strict mode.
             val vcsUrlNoCredentials = target.vcsProcessed.url.stripCredentialsFromUrl()
             if (vcsUrlNoCredentials != target.vcsProcessed.url) {
-                // Try once more with any user name / password stripped from the URL.
+                // SSH always requires authentication, so it does not make sense to try without. Instead try HTTPS.
```

Comment:
The credentials could be configured in the SSH config. Also, the test cases all relate to Gerrit, and in Gerrit replacing the protocol doesn't work, because HTTPS uses a different URL, e.g. `https://example.com/a/repo` instead of `ssh://example.com:29418/repo`.

Reply:
Right. So a more generic approach would be to allow the configuration of a Regex-like search-and-replace pattern for URLs (I believe it was actually you who suggested that). I'll look into that.